tree/dataframe/test/dataframe_regression_tests.hxx

```diff
@@ -106,3 +109,100 @@ TEST(TEST_CATEGORY, UniqueEntryNumbers)
 
    gSystem->Unlink(fname);
 }
+
+// ROOT-9731
+TEST(TEST_CATEGORY, ReadVector3)
+{
+   const std::string filename = "readwritetvector3.root";
+   {
+      TFile tfile(filename.c_str(), "recreate");
+      TTree tree("t", "t");
+      TVector3 a;
+      tree.Branch("a", &a); // TVector3 as TBranchElement
```

Comment:
Is it really a TBranchElement?  TVector3 is not splittable (It has a custom streamer).

Reply:
yes, that's easily verified:

```
root [0] TTree tree("t", "t");
root [1] TVector3 a;
root [2] tree.Branch("a", &a); 
root [3] tree.GetBranch("a")->ClassName()
(const char *) "TBranchElement"
```